core/state/statedb_arbitrum.go

```diff
@@ -272,9 +311,9 @@ func (s *StateDB) RecordProgram(targets []rawdb.WasmTarget, moduleHash common.Ha
 		// nothing to record
 		return
 	}
-	asmMap, err := s.TryGetActivatedAsmMap(targets, moduleHash)
-	if err != nil {
-		log.Crit("can't find activated wasm while recording", "modulehash", moduleHash, "err", err)
+	asmMap, missingTargets, err := s.ActivatedAsmMap(targets, moduleHash)
+	if err != nil || len(missingTargets) > 0 {
+		log.Crit("can't find activated wasm while recording", "modulehash", moduleHash, "err", err, "missingTargets", missingTargets)
```

Comment:
log.Crit means we exit the node, correct? If so, it would be nice to give a hint to solve the issue.

Reply:
yes, log.Crit calls os.Exit and it may corrupt databases

after log.Crit there's no good way to recover, except from restoring node from a snapshot, and that applies to all log.Crits

this check should fail only in case of a bug, so maybe it's not that bad to kill the node here

alternatively we could log.Error here and the issue of missing targets would propagate to validation workers that would fail because of the missing targets

Reply:
try to nevere use log.Crit.
Using panic is much better, because it'll only kill an rpc if we're inside such context.
That's only if you cannot return an error, of course.

Reply:
from an initial look - it seems like we can change the function to return an error here